catalyst_voices/lib/common/formatters/date_formatter.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:catalyst_voices_shared/src/utils/date_time_ext.dart';
+import 'package:intl/intl.dart';
+
+/// A [DateTime] formatter.
+abstract class DateFormatter {
+  /// Formats recent dates like:
+  /// - Today
+  /// - Tomorrow
+  /// - Yesterday
+  /// - 2 days ago
+  /// - Other cases: yMMMMd date format.
+  static String formatRecentDate(VoicesLocalizations l10n, DateTime dateTime) {
+    final now = DateTime.now();
```

Comment:
This is not really testable as we can't mock `DateTime.now()`.

What do you think about having such class? Maybe it should be just abstract final class `MockableDateTime`.
```dart
extension DateTimeExt on DateTime {
  static DateTime? _mockedDateTime;

  @visibleForTesting
  static set mockedDateTime(DateTime? customTime) {
    _mockedDateTime = customTime;
  }

  /// Testable [DateTime] factory method which returns:
  /// - mocked value, if not null, set with [mockedDateTime]
  /// - if[utc] current utc [DateTime.timestamp]
  /// - else current local [DateTime.now]
  static DateTime now({
    bool utc = false,
  }) {
    DateTime? getMockedDateTime() {
      return utc ? _mockedDateTime?.toUtc() : _mockedDateTime;
    }

    DateTime getDateTime() {
      return utc ? DateTime.timestamp() : DateTime.now();
    }

    return getMockedDateTime() ?? getDateTime();
  }
}
```

Another approche, less invasive, it to have optional named parameter like so.

```dart
static String formatRecentDate(VoicesLocalizations l10n, DateTime dateTime, {Datetime? from})

...
from ??= DateTime.now();

```

Reply:
Alright, I added the approach with DateTimExt.now() mockable factory.